usmqe/api/base.py

```diff
@@ -0,0 +1,104 @@
+
+"""
+Basic REST API.
+"""
+
+import pytest
+import json
+
+LOGGER = pytest.get_logger("api_base", module=True)
+
+
+class ApiBase(object):
+    """ Basic class for REST API.
+    """
+
+    default_asserts = {
+        "cookies": None,
+        "ok": True,
+        "reason": 'OK',
+        "status": 200,
+    }
+
+    @staticmethod
+    def print_req_info(resp):
+        """ Print debug information.
+
+        Args:
+            resp: response
+        """
+        LOGGER.debug("request.url:  %s" % resp.request.url)
+        LOGGER.debug("request.method:  %s" % resp.request.method)
+        LOGGER.debug("request.body:  %s" % resp.request.body)
+        LOGGER.debug("request.headers:  %s" % resp.request.headers)
+        LOGGER.debug("response.cookies: %s" % resp.cookies)
+        LOGGER.debug("response.content: %s" % resp.content)
+        LOGGER.debug("response.headers: %s" % resp.headers)
+        try:
+            LOGGER.debug(
+                "response.json:    %s" % resp.json(encoding='unicode'))
+        except ValueError:
+            LOGGER.debug("response.json:    ")
+        LOGGER.debug("response.ok:      %s" % resp.ok)
+        LOGGER.debug("response.reason:  %s" % resp.reason)
+        LOGGER.debug("response.status:  %s" % resp.status_code)
+        LOGGER.debug("response.text:    %s" % resp.text)
+
+    @staticmethod
+    def check_response(resp, asserts_in=None):
+        """ Check default asserts.
+
+        It checks: *ok*, *status*, *reason*.
+        Args:
+            resp: response to check
+            asserts_in: asserts that are compared with response
+        """
+
+        asserts = ApiBase.default_asserts.copy()
+        if asserts_in:
+            asserts.update(asserts_in)
+        # if "cookies" in asserts and asserts["cookies"] is None:
+        #    pytest.check(not(resp.cookies), "Cookies should be empty.")
+        try:
+            json.dumps(resp.json(encoding='unicode'))
+        except ValueError:
+            pytest.check(False, issue="Bad response json format.")
+        pytest.check(
+            resp.ok == asserts["ok"],
+            "There should be ok == %s." % str(asserts["ok"]))
```

Comment:
The ugly think related to this is, that logging functions are still using the old `%` style for formatting.
I think, that for example you should use 
```
LOGGER.info("Foo: %s", variable)
```
not
```
LOGGER.info("Foo: {}".format(variable))
```
But I agree, that on other places, we should use the new `format` style preferred for python 3.